File: LeetCode.com/problems/deleteColumnsToMakeSorted.js
/**
 * @param {string[]} strs
 * @return {number}
 */
const minDeletionSize = (strs) => {
  let result = 0;

  for (let i = 0; i < strs[0].length; i++) {
    let curr = '';
    for (let j = 0; j < strs.length; j++) {
      curr += strs[j][i];
    }
    if (curr != curr.split('').sort().join('')) result++;
  }

  return result;
};

console.log(minDeletionSize(['rrjk', 'furt', 'guzm']));
console.log(minDeletionSize(['abc', 'bce', 'cae']));
console.log(minDeletionSize(['a', 'b']));

File: LeetCode.com/problemsCpp/deleteColumnsToMakeSorted.cpp
#include <algorithm>
#include <iostream>
#include <string>
#include <vector>

using namespace std;

class Solution {
 public:
  int minDeletionSize(vector<string>& strs) {
    int result = 0;

    for (int i = 0; i < strs[0].size(); i++) {
      for (int j = 0; j < strs.size() - 1; j++) {
        if (strs[j][i] > strs[j + 1][i]) {
          result++;
          break;
        }
      }
    }

    return result;
  }
};

int main() {
  vector<string> strs;

  strs.push_back("abc");
  strs.push_back("bce");
  strs.push_back("cae");

  Solution sol;

  cout << sol.minDeletionSize(strs) << endl;
}
